source/material_model/rheology/composite_visco_plastic.cc

```diff
@@ -70,11 +72,59 @@ namespace aspect
       double
       CompositeViscoPlastic<dim>::compute_viscosity (const double pressure,
                                                      const double temperature,
-                                                     const unsigned int composition,
+                                                     const std::vector<double> &volume_fractions,
                                                      const SymmetricTensor<2,dim> &strain_rate,
                                                      std::vector<double> &partial_strain_rates,
                                                      const std::vector<double> &phase_function_values,
                                                      const std::vector<unsigned int> &n_phases_per_composition) const
+      {
+        double viscosity = 0.;
+        partial_strain_rates.resize(5, 0.);
+
+        // Isostrain averaging
+        double total_volume_fraction = 1.;
+        for (unsigned int composition=0; composition < number_of_compositions; ++composition)
+          {
+            // Only include the contribution to the viscosity
+            // from a given composition if the volume fraction exceeds
+            // a certain (small) fraction.
+            if (volume_fractions[composition] > 1.e-5)
```

Comment:
Why not set this value to 0?

Reply:
My initial thought was to speed up calculations where one or more volume fractions were small-ish, but you're right, that might introduce (small) unexpected variations. I've now set it to 2.*std::numeric_limits<double>::epsilon().